presto-orc/src/main/java/com/facebook/presto/orc/writer/DictionaryColumnWriter.java

```diff
@@ -410,11 +408,15 @@ public long getBufferedBytes()
     @Override
     public long getRetainedBytes()
     {
-        return sizeOf(rowGroupIndexes) +
+        long retainedBytes = sizeOf(rowGroupIndexes) +
                 dataStream.getRetainedBytes() +
                 presentStream.getRetainedBytes() +
-                getRetainedDictionaryBytes() +
-                columnStatisticsRetainedSizeInBytes;
+                getRetainedDictionaryBytes();
+        for (DictionaryRowGroup rowGroup : rowGroups) {
+            retainedBytes += rowGroup.getColumnStatistics().getRetainedSizeInBytes();
```

Comment:
nit: I would recommend adding a `DictionaryRowGroup#getRetainedSizeInBytes` method that accounts both `dictionaryIndexes`, `columnStatistics` and the size of the `DictionaryRowGroup` instance itself and using it for memory accounting

Reply:
+1, let us also track the total size when a row group is finished in finishRowGroup instead of computing it every time. After a row group is finished it is immutable.